core/src/services/azblob/backend.rs

```diff
@@ -545,7 +548,17 @@ impl Access for AzblobBackend {
         let status = resp.status();
 
         match status {
-            StatusCode::OK => parse_into_metadata(path, resp.headers()).map(RpStat::new),
+            StatusCode::OK => {
+                let headers = resp.headers();
+                let mut meta = parse_into_metadata(path, headers)?;
+
+                let user_meta = parse_prefixed_headers(&headers, X_MS_META_NAME_PREFIX);
+                if !user_meta.is_empty() {
```

Comment:
This check is also done in the s3 backend. 

Should we always call `meta.with_user_metadata(user_meta);` instead of checking if the `user_meta` is empty?

Or can we for example, add that checking inside `meta.with_user_metadata(user_meta);` and do an early return if the `user_meta` hashmap is empty?https://github.com/apache/opendal/blob/19279327a73ffa70f4c5079fd149ab82f7649e15/core/src/types/metadata.rs#L524

I find that checking in each backend if the hashmap is empty is error-prone boilerplate...

Reply:
It's fine to have such a check. OpenDAL tends to have more readable code because we have many services, and that code is rarely modified.